builtin/clone.c

```diff
@@ -929,19 +931,19 @@ static int path_exists(const char *path)
 	return !stat(path, &sb);
 }
 
-/**
- * Read config variables.
- */
-static int git_pull_config(const char *var, const char *value, void *cb)
-{
-	if (!strcmp(var, "submodule.recurse")) {
-		option_recurse_submodules.nr = git_config_bool(var, value) ?
-			1 : option_recurse_submodules.nr;
-		return 0;
-	}
-	return git_default_config(var, value, cb);
-}
-
+/**
+ * Read config variables.
+ */
+static int git_clone_config(const char *var, const char *value, void *cb)
+{
+	if (!strcmp(var, "submodule.recurse")) {
+		option_recurse_submodules.nr = git_config_bool(var, value) ?
+			RECURSE_SUBMODULES_ON : RECURSE_SUBMODULES_OFF;
```

Comment:
This does way more than just rename the function. And it still modifies `nr` without allocating/initializing the item. The correct thing to do here is more likely to be this:

```c
string_list_append(&option_recurse_submodules, ".");
```

(compare to what `recurse_submodules_cb()` does when no `arg` is provided)